decode.go

```diff
@@ -30,8 +32,8 @@ func decodedLen(src []byte) (blockLen, headerLen int, err error) {
 	if n <= 0 {
 		return 0, 0, ErrCorrupt
 	}
-	if uint64(int(v)) != v {
-		return 0, 0, errors.New("snappy: decoded block is too large")
+	if uint64(uint32(v)) != v {
```

Comment:
I'll push the simpler check (against 0xffffffff), although I think it should be `>`, since 2^32-1 is still a valid length.

I agree about the signed/unsigned slice length problems. Maybe we need to say that slices >2G can't be uncompressed on 32-bit platforms via the `Decode()` API?  Using the `Reader()` interface should still be fine though.